Realm/RLMSyncUser.h

```diff
@@ -113,8 +140,118 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
+ */
+- (nullable RLMNotificationToken *)modifyRealmPermissionsForURL:(nullable NSURL *)realmURL
+                                                   userIdentity:(nullable NSString *)identity
+                                                    permissions:(RLMSyncRealmPermission)permissions
+                                                       callback:(nullable RLMSyncPermissionChangeBlock)callback
+NS_REFINED_FOR_SWIFT;
+
+/**
+ Create a permission offer.
+
+ A permission offer is repesented by a portable string token which can be
+ given to another user through any convenient means (such as email). It
+ represents offering certain permissions to the recepient, allowing them
+ to work with a synced Realm managed by this user.
+
+ The token itself is returned asynchronously through the `callback` block.
+ It is this token string which can be sent to a different user and then
+ passed into `-acceptOffer:callback:` to accept the offer. Do not confuse
+ this token with the identifier returned immediately by this method.
+
+ @param realmURL        The URL of the synced Realm on the server.
+ @param date            The date when the offer should expire. If nil, the
+                        offer will never expire.
+ @param permissions     The permissions that the offer should include.
+ @param token           A pointer to a notification token object, through
+                        which a notification token should be passed out.
+ @param callback        A block through which the status of the permission
+                        offer creation process should be reported.
+ @return    An identifier for the newly-created permission offer. This
+            identifier can be passed into any method taking an `offerKey`
+            parameter. Do not confuse this with the permission offer's
+            token, which is passed back asynchronously through the callback
+            once the server has processed the request.
+ */
+- (NSString *)offerPermissionsForURL:(NSURL *)realmURL
+                          expiration:(nullable NSDate *)date
+                         permissions:(RLMSyncRealmPermission)permissions
+                               token:(RLMNotificationToken * _Nullable __autoreleasing * _Nonnull)token
+                            callback:(RLMSyncOfferChangeBlock)callback
+NS_REFINED_FOR_SWIFT;
+
+// TODO: does this really need a callback? Won't the object just be deleted immediately?
+/**
+ Revoke a permission offer which was previously generated.
+
+ @param offerKey        The identifier of the previously-generated permission
+                        offer (not the token).
+ @param callback        An optional callback block, to asynchronously report
+                        the progress of the operation.
+ @return    If a callback was specified, this will be a Realm notification
+            token. Call `-stop` if you want to deactivate the callback.
+ */
+- (nullable RLMNotificationToken *)deleteOffer:(NSString *)offerKey
```

Comment:
Per my comment above, I think this should accept a permission offer token instead of a key.